### ISSUE TITLE
[Vite Bundle] Ability to not have leading slash in base

### ISSUE THREAD

**Issue by andyexeter** [lhapaipai/symfony-vite-dev#7]
Hey there,

Not sure if I should be creating an issue for the Symfony Bundle here or in the subtree split repo. Let me know if you want me to move it :)

In the  be great if there was a way to stop the base option in `configs` being prefixed with a forward slash here:

https://github.com/lhapaipai/symfony-vite-dev/blob/cd1e3a5fb6cc3378e07af46bc11b30e6a4efdf2d/src/vite-bundle/src/DependencyInjection/PentatrionViteExtension.php#L162

The reason I'd like this is because it assists with subdirectory deploys. For example when a web application's document root is at e.g. `/var/www/example.org/some/deep/subdirectory` and is served via https://example.org/some/deep/subdirectory

When [Symfony's PathPackage](https://github.com/symfony/symfony/blob/dc330b087e496d2851cf9461fe27b16e69a66c89/src/Symfony/Component/Asset/PathPackage.php) encounters an asset which is not absolute and does not start with a trailing slash e.g. `build/images/my-image.jpg`, it prefixes the URL with the current request base path to allow subdirectory deploys, so `https://example.org/some/deep/subdirectory/build/images/my-image.jpg` is returned which is the correct URL for the resource.

The problem is, by the time PathPackage receives the URL it's been processed by the `ViteAssetVersionStrategy` which always prepends a forward slash to the URI.

I've had a look around the code and it appears the base config option is being used for file paths e.g. in FileAccessor as well as asset URIs:

https://github.com/lhapaipai/symfony-vite-dev/blob/cd1e3a5fb6cc3378e07af46bc11b30e6a4efdf2d/src/vite-bundle/src/Service/FileAccessor.php#L35-L40

So it's not a simple case of removing the slash. I suspect we would need to split out the config into two options, one for paths and one for URIs. What do you think?

**Comment by lhapaipai** [comment]
Hi @andyexeter, I understand the issue but to be sure to correct the problem could you highlight your situation in a fork of the repository inside the `playground/basic` or in a new playground ?
Thank you very much

**Comment by andyexeter** [comment]
Hi @lhapaipai,

I'm having some trouble getting the docker playground running. It keeps throwing the following error when I try to load https://docker.symfony-vite-dev.localhost/:

```
Attempted to load class "PentatrionViteBundle" from namespace "Pentatrion\ViteBundle".
Did you forget a "use" statement for another namespace?
```

![Screenshot from 2024-01-02 13-07-27](https://github.com/lhapaipai/symfony-vite-dev/assets/6660584/48e741eb-19f5-4c64-a9d4-4dd1ff8499b1)

I think this is happening because composer is symlinking the vite-bundle package to src/vite-bundle but this path does not exist within the php docker container:

```console
$ docker compose exec php sh
/srv/app # cd vendor/pentatrion/
/srv/app/vendor/pentatrion # ls -la
total 8
drwxrwxr-x    2 1000     1000          4096 Jan  2 13:02 .
drwxrwxr-x   10 1000     1000          4096 Jan  2 13:02 ..
lrwxrwxrwx    1 1000     1000            28 Jan  2 13:02 vite-bundle -> ../../../../src/vite-bundle/

/srv/app/vendor/pentatrion # file ../../../../src/vite-bundle/
../../../../src/vite-bundle/: cannot open `../../../../src/vite-bundle/' (No such file or directory)


```

**Comment by lhapaipai** [comment]
hi @andyexeter,
yes you're right. you can see that the vite-bundle directory has been bind mounted in the `compose.override.yml` file
https://github.com/lhapaipai/symfony-vite-dev/blob/c0285995bc0ec32128918dc3d6ce9c5f2cc1f990/playground/docker/compose.override.yml#L7
but the link from `composer.json` is wrong.
I fixed this.

```json
{
    "repositories": [
        {
            "type": "path",
            "url": "/extra/vite-bundle"
        }
    ]
}
```

**Comment by andyexeter** [comment]
HI @lhapaipai,

Sorry for the delay, I've got a working reproducer set up now at my fork: https://github.com/andyexeter/symfony-vite-dev/tree/main/playground/docker-subdirectory

Set up is as follows:

1. Clone and `cd` into the fork
2. Run `make install` and `make install-playgrounds`
3. `cd playground/docker-subdirectory` and run the playground as per the instructions in `playground/docker-subdirectory/README.md`
4. Run `make vite-build` to build for production
5. Access the site via https://docker-subdirectory.symfony-vite-dev.localhost/public/

You'll see that the kitten.jpg image fails to load. This is happening because of the reason I mentioned above where `ViteAssetVersionStrategy` always prepends a forward slash so `PathPackage` doesn't prepend the base path:

https://github.com/symfony/symfony/blob/7008c1ddc1ca558a7ef8f29b4cb6b2d34467e279/src/Symfony/Component/Asset/PathPackage.php#L53-L55:

```php
// if absolute or begins with /, we're done
if ($this->isAbsoluteUrl($versionedPath) || ($versionedPath && '/' === $versionedPath[0])) {
    return $versionedPath;
}
```

As a quick proof of concept, you can uncomment this line in my fork: https://github.com/andyexeter/symfony-vite-dev/blob/dd9e03086941b5f2a6fb38cf4c50a94c7c3cdc31/src/vite-bundle/src/Asset/ViteAssetVersionStrategy.php#L85

You'll see that the kitten.jpg image then loads because the forward slash is no longer prepended and PathPacakge correctly prepends the base path, changing the image src from `/build/assets/kitten-O01njDza.jpg` to `/public/build/assets/kitten-O01njDza.jpg`.

So, in my opinion the solution to this is to split the current basePath property in `ViteAssetVersionStrategy` in two - one value for URIs and the other for files.

I have been looking into this a bit more and I think the `absolute_url` option _could_ fix the problem, however the `completeURL` method is currently ignoring base path and port from the URL to create an absolute URL.

If we set `absolute_url: true` and the site is served from http://example.org:8080/subdirectory`, the `completeURL` method will currently omit the `:8080` and `/subdirectory` part from the returned absolute URL.

I hope this makes sense :)

**Comment by lhapaipai** [comment]
fixed with #13
